### DISCUSSION TITLE
Input on my recent topology

### DISCUSSION THREAD

**Discussion by Astrotokii** [ipspace/netlab#1537] [Show and tell]
Im a CCNA student and I tried my hand at a multiarea OSPF with redundancy. If you have a minute to look and give me feedback, I would greatly appreciate it! I am about a month into learning networking.

https://github.com/Astrotokii/NetlabLabs/tree/6352c4a8cf1f813a353b448416052fc3a8af5ea8/OSPF_M_B_P2P

**Comment by ipspace** [comment]
> Im a CCNA student

Welcome! So nice to hear someone found netlab useful for certification preparation

> and I tried my hand at a multiarea OSPF with redundancy.

I'm impressed. Pretty large one ;) Can I post a link to your GitHub directory?

> If you have a minute to look and give me feedback

You did a really good job. I love how you used groups to simplify the setup. Also, using "name" node attribute is creative (it does get overwritten, but it's a nice documentation touch). The only thing that I found a bit odd was using "0.0.0.0" as router ID. I'm guessing it works, but I don't think I've ever seen anyone doing it that way.

**Reply by Astrotokii** [reply]
Yeah you can absolutely post a link :) 

The main reason I used 0.0.0.0 was to keep the router IDs similar to the area #, not sure if itd work in a real situation haha

**Reply by jbemmel** [reply]
I found https://community.cisco.com/t5/other-network-architecture-subjects/ospf-router-id-0-0-0-0/td-p/275142 from 2004

The router IDs should get auto-generated, there is no need to set them manually (but you can if you like)

Similar for some of the other attributes you set - it might be interesting to test which configurations can be removed while still ending up with a working system

**Reply by Astrotokii** [reply]
Yeah, I know I can leave out the 0.0.0.0 area specifications because they are cisco default and most the link ospf.area specifications since I set a default value for them in nodes, however, for me to keep track I just redundantly added them. Im legally blind so following trails of not explicitly stated paths causes a lot of unneeded eye strain. Also hence the node naming; the _a# specifying the ospf area so I dont have to draw a map out hahah

**Comment by DanPartelly** [comment]
You are doing very well.  Now that you done some topologies, and big ones at that, how do you find working with netlab ?

> I am about a month into learning networking.
> 
> https://github.com/Astrotokii/NetlabLabs/tree/6352c4a8cf1f813a353b448416052fc3a8af5ea8/OSPF_M_B_P2P

**Reply by Astrotokii** [reply]
Honestly, I started out with cisco packet tracer and gns3. I hated them haha. I saw someone mention netlab in a reddit thread, so I checked it out. Lets just say, I bought a new laptop that had windows 11 pro, nuked that for ubuntu, all because I wanted to use netlab over everything else. Netlab literally just works for my brain, cant explain it.

(Theres an issue using multipass on windows 11 pro, because hyper-v roadblocks nested virtualization)

**Reply by jbemmel** [reply]
It’s the data model approach, it forms a solid conceptual model in your brain and then the rest logically follows from that

I would encourage you to explore other platforms beyond IOS - working multi-vendor is like speaking different languages. It helps you appreciate the good parts and the limitations/trade-offs for different design choices and quirks

**Reply by Astrotokii** [reply]
Great advice, I had started exploring SR linux and FRR as well. I was also planning on doing Juniper as well once I got the basic networking stuff down.

**Reply by DanPartelly** [reply]
Im glad netlab clicked with you. It's an awesome program, with a lot of utility.

In case you didn't know this place already, here is a link towards [Junpier's Day One library](https://www.juniper.net/documentation/jnbooks/us/en/day-one-books) While they do use Juniper gear to illustrate the concepts, that library contains lot of general networking gems and concepts, very well explained. And it's all free.  

>  Netlab literally just works for my brain, cant explain it.
> 
> (Theres an issue using multipass on windows 11 pro, because hyper-v roadblocks nested virtualization)
